src/components/v2/appDetails/ea/EAAppDetail.component.tsx

```diff
@@ -77,13 +86,42 @@ function ExternalAppDetail({appId, appName}) {
         return genericAppDetail
     }
 
-    const _getAndSetAppDetail = () => {
+    const _getAndSetAppDetail = async () => {
         isAPICallInProgress = true;
         getAppDetail(appId)
             .then((appDetailResponse: HelmAppDetailResponse) => {
                 IndexStore.publishAppDetails(_convertToGenericAppDetailModel(appDetailResponse.result));
-                setIsLoading(false);
-                isAPICallInProgress = false;
+
+                if (appDetailResponse.result?.appDetail.environmentDetails.clusterId) {
+                    Promise.all([
+                        getMonitoringTools(),
+                        getExternalLinks(appDetailResponse.result.appDetail.environmentDetails.clusterId),
+                    ])
+                        .then(([monitoringToolsRes, externalLinksRes]) => {
+                            setExternalLinksAndTools({
+                                externalLinks: externalLinksRes.result?.sort(sortByUpdatedOn) || [],
+                                monitoringTools:
+                                    monitoringToolsRes.result
+                                        ?.map((tool) => ({
+                                            label: tool.name,
+                                            value: tool.id,
+                                            icon: tool.icon,
+                                        }))
+                                        .sort(sortOptionsByValue) || [],
+                            })
+                            setIsLoading(false)
+                            isAPICallInProgress = false
+                        })
+                        .catch((e) => {
+                            setExternalLinksAndTools(externalLinksAndTools)
```

Comment:
no error handling?

Reply:
As discussed, for error handling setting the old list instead of halting the page load or showing any error for this particular section on the page.